core/silkworm/execution/evm.cpp

```diff
@@ -249,9 +249,10 @@ evmc::result EVM::execute(const evmc_message& msg, ByteView code, std::optional<
     if (exo_evm) {
         EvmHost host{*this};
         res = exo_evm->execute(exo_evm, &host.get_interface(), host.to_context(), rev, &msg, code.data(), code.size());
-    } else if (analysis_cache) {
+    } else if (code_hash != std::nullopt && advanced_analysis_cache != nullptr) {
```

Comment:
When is `advanced_analysis_cache` not null?

Reply:
It's not null for mainnet block execution (`cmd/execute`), but it's null for the consensus tests (`cmd/consensus`).

I haven't implemented analysis cache for baseline interpreter yet (on my TODO list). Then it'll be interesting to compare baseline & advanced interpreter for mainnet block execution when they both have analysis caching. When I compare advanced interpreter with analysis cache against baseline interpreter w/o one, advanced interpreter wins.